packages/@aws-cdk/aws-dynamodb/lib/replica-provider.ts

```diff
@@ -27,10 +27,36 @@ export class ReplicaProvider extends NestedStack {
    */
   public static getOrCreate(scope: Construct, props: ReplicaProviderProps = {}) {
     const stack = Stack.of(scope);
-    const uid = '@aws-cdk/aws-dynamodb.ReplicaProvider';
+    const uid = this.getUid(stack);
     return stack.node.tryFindChild(uid) as ReplicaProvider ?? new ReplicaProvider(stack, uid, props);
   }
 
+  public static clearGetOrCreateCalls() {
+    this.getOrCreateCalls.clear();
+  }
+
+  // Map of getOrCreate() calls per stack
+  private static getOrCreateCalls = new Map<string, number>();
+
+  private static getUid(stack: Stack): string {
+    // The custom resource implementation uses IAM managed policies. There's
+    // a limit of 10 managed policies per role in IAM. So we create a new
+    // provider if we reach this limit.
+    const MAX_MANAGED_POLICIES = 10;
+    let uid = '@aws-cdk/aws-dynamodb.ReplicaProvider';
+    const calls = this.getOrCreateCalls.get(stack.stackName);
+    if (!calls) {
+      this.getOrCreateCalls.set(stack.stackName, 1);
+    } else {
+      if (calls >= MAX_MANAGED_POLICIES) {
```

Comment:
Am I reading this right that the new max would become 20?  I realize not common, but very easily possible to have a stack with over 20 table definitions.  Can this be generalized for every group of 10?

Reply:
No, there's no new max: every 10 managed policy we create a new nested stack. See the unit test that creates 21 tables.